operations.go

```diff
@@ -9,60 +9,73 @@ type operation func()
 
 // Operations is a task executor.
 type operations struct {
-	ops     []operation
-	mu      sync.Mutex
-	startMu sync.Mutex
-	done    chan struct{}
+	mu   sync.Mutex
+	busy bool
+	ops  []operation
 }
 
 func newOperations() *operations {
 	closed := make(chan struct{})
 	close(closed)
-	return &operations{
-		done: closed,
-	}
+	return &operations{}
 }
 
 // Enqueue adds a new action to be executed. If there are no actions scheduled,
 // the execution will start immediately in a new goroutine.
 func (o *operations) Enqueue(op operation) {
+	if op == nil {
+		return
+	}
+
 	o.mu.Lock()
-	defer o.mu.Unlock()
+	busy := o.busy
```

Comment:
What do you think of something like `queueInProgress`? Major nit, but for some reason I kept having to re-read because `busy` is on the struct and on stack.